### PR TITLE
Change layer name

### LINKED ISSUE #8246: Need to change the name of the layers

### PR DESCRIPTION
**Pull request by nobuti**
This PR fixes #8246.

rename

Also, it implements the deletion of the layer in the inner view:

delete-layer

cc @xavijam @piensaenpixel

### CODE REVIEW

**Comment by xavijam** [comment]
Could we add, in this state, the modal confirmation for layer remove? please?

**Comment by saleiva** [comment]
Ooooh! 
Also, can we make double click on layer name to activate the editting box?

**Comment by saleiva** [comment]
BTW this is ❤️

**Comment by nobuti** [comment]
Confirmation and double click done. But this last one is a bit tricky, at least in the layers list view, because we are adding click (to edit the layer) and double click (to edit the layer's name) event to the same element. I had to add some delay (250ms) to segment those actions.

**Comment by nobuti** [comment]
confirmation

**Comment by xavijam** [comment]
Something I've done in the table with the rename is, create an '<input>' with the table name, make it readonly and only enable it when we need to edit it. How does it sound?

File: lib/assets/javascripts/cartodb3/editor/layers/layer-views/data-layer-view.js [hunk 1]
```diff
@@ -145,6 +161,83 @@ module.exports = CoreView.extend({
...
+    this._modals.create(function (modalModel) {
+      return new ConfirmationView({
+        modalModel: modalModel,
+        template: templateConfirmation,
+        renderOpts: {
+          layerName: self.model.getTableName()
+        },
+        runAction: function () {
+          self.model.destroy();
+          modalModel.destroy();
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-views/data-layer-view.js]: I'd not close the modal until the action is completed, successfully or mistakenly.

File: lib/assets/javascripts/cartodb3/editor/layers/layer-views/layer-rename.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  var newName = opts.newName;
+  var oldName = layerDefinitionModel.getName();
+
+  var notification = Notifier.addNotification({
+    status: 'loading',
+    info: _t('editor.layers.rename.loading'),
+    closable: false
+  });
+
+  layerDefinitionModel.set({table_name_alias: newName}, {silent: true});
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-views/layer-rename.js]: Do we need to silent it?

**Comment by saleiva** [comment]
Do we even have notifier events?

**Comment by xavijam** [comment]
Of course.

**Comment by saleiva** [comment]
vamosnenos

File: lib/assets/javascripts/cartodb3/editor/layers/layer-views/data-layer-view.js (hunk 1 above)
Comment (nobuti) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-views/data-layer-view.js]: But If we don't close it the notification happens in the background, below the modal, doesn't it?

Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-views/data-layer-view.js]: Ok, deal, we have the loading in the modal view but better to have the notifier 👍

File: lib/assets/javascripts/cartodb3/editor/components/inline-editor/inline-editor-view.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
+var CoreView = require('backbone/core-view');
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/components/inline-editor/inline-editor-view.js]: Peeeeeerfect, could we move this component to the general components please? I'll use it in the dataset view. ❤️

**Comment by nobuti** [comment]
Done.

**Comment by xavijam** [comment]
Ready then?

**Pull request merged**